Dolphin scripts/Entrance Randomizer/lib/utils.py

```diff
@@ -24,6 +25,11 @@ class State:
     current_area_new = 0
     """Area ID of the current frame"""
     visited_altar_of_ages = False
+    visited_spirits: ClassVar[dict[int, bool]] = {
+        LevelCRC.MONKEY_SPIRIT: False,
+        LevelCRC.SCORPION_SPIRIT: False,
+        LevelCRC.PENGUIN_SPIRIT: False,
+    }
```

Comment:
Suggestion to generalise this with a `set[int]` of "visited levels", we can add the current level in here indiscriminately to simplify the runtime logic a bit and be more flexible for future special cases. (can be done as a follow-up)
```suggestion
    visited_levels = set[int]()
```